File: Sprint8/B/C++/main.cpp
// https://contest.yandex.ru/contest/26131/problems/B/
#include <iostream>
#include <string>
#include <cmath>

int main() {
    std::string passport_name, base_name;
    std::cin >> passport_name >> base_name;

    int diff = passport_name.length() - base_name.length();
    if (std::abs(diff) > 1) {
        std::cout << "FAIL" << std::endl;
        return 0;
    }

    int i = 0, j = 0;
    bool insertion = false, deletion = false, change = false;
    while (i < passport_name.length() && j < base_name.length()) {
        if (passport_name[i] != base_name[j]) {
            if (!insertion && !deletion && !change && passport_name.length() < base_name.length()) {
                insertion = true;
                j++;
            } else if (!insertion && !deletion && !change && passport_name.length() > base_name.length()) {
                deletion = true;
                i++;
            } else if (!insertion && !deletion && !change && passport_name.length() == base_name.length()) {
                change = true;
            } else {
                std::cout << "FAIL" << std::endl;
                return 0;
            }
        }

        i++;
        j++;
    }

    std::cout << "OK" << std::endl;
    return 0;
}

File: Sprint8/B/Rust/src/main.rs
// https://contest.yandex.ru/contest/26131/problems/B/
use std::io::BufRead;

fn can_transform(source: &str, target: &str) -> bool {
    let diff = source.len() as isize - target.len() as isize;

    if diff.abs() > 1 {
        return false;
    }

    let mut source_chars = source.chars();
    let mut target_chars = target.chars();
    let mut change_allowed = true;

    loop {
        match (source_chars.next(), target_chars.next()) {
            (Some(sc), Some(tc)) if sc == tc => continue,
            (Some(_), Some(_)) if change_allowed => {
                change_allowed = false;
                if diff == 0 {
                    continue;
                } else if diff == 1 {
                    source_chars.next();
                    continue;
                } else if diff == -1 {
                    target_chars.next();
                    continue;
                }
            }
            (Some(_), None) if diff == 1 && change_allowed => return true,
            (None, Some(_)) if diff == -1 && change_allowed => return true,
            (None, None) => return true,
            _ => return false,
        }
    }
}

fn main() {
    let mut lines = std::io::stdin().lock().lines();

    let passport_name = lines.next().unwrap().unwrap();
    let base_name = lines.next().unwrap().unwrap();

    match can_transform(&passport_name, &base_name) {
        true => println!("OK"),
        false => println!("FAIL"),
    }
}
